### ISSUE TITLE
Opening new window while in zen mode causes it to use centered layout

### ISSUE THREAD

**Issue by jeffhube** [Microsoft/vscode#46277]
Issue Type: <b>Bug</b>

1. Activate zen mode (Command Palette > Toggle Zen Mode)
2. Open a new window (Command Palette > New Window)

Expected behavior: the new window is *not* using the zen mode centered layout
Actual behavior: the new window is using the zen mode centered layout

This seems similar to https://github.com/Microsoft/vscode/issues/45610 , except vscode isn't being closed.

VS Code version: Code 1.21.1 (79b44aa704ce542d8ca4a3cc44cfca566e7720f1, 2018-03-14T14:46:47.128Z)
OS version: Windows_NT x64 10.0.16299

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7600U CPU @ 2.80GHz (4 x 2904)|
|Memory (System)|15.89GB (8.61GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (14)</summary>

Extension|Author (truncated)|Version
---|---|---
xml|Dot|1.9.2
tslint|eg2|1.0.28
python|ms-|2018.2.1
cpptools|ms-|0.15.0
csharp|ms-|1.14.0
vetur|oct|0.11.7
typescript-hero|rbb|2.3.1
rust|rus|0.4.0
salesforcedx-vscode|sal|42.7.0
salesforcedx-vscode-apex|sal|42.7.0
salesforcedx-vscode-apex-debugger|sal|42.7.0
salesforcedx-vscode-core|sal|42.7.0
salesforcedx-vscode-lightning|sal|42.7.0
salesforcedx-vscode-visualforce|sal|42.7.0

</details>
Reproduces without extensions

**Comment by isidorn** [comment]
We decided for th ecntered editor layour to be preserved across workspaces https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/browser/parts/editor/editorGroupsControl.ts#L1059

This is just a design decision we took. So let's keep this issue open and wait for more feedback to potentially change that

fyi @SrTobi

**Comment by SrTobi** [comment]
No, I think @jeffhube is right... the problem is not that the centered editor is stored in `StorageScope.GLOBAL` but that it is stored at all when it was activated by transitioning to the zen mode. It's actually quite similar to the problems I mentioned in #45610 and #45773.

The main problem is the following: when we transition to zenmode we change all kind of things (we toggle the centered layout, we hide the sidebar etc) and we do it in a way that the user thinks it's temporal (which it kind of is, because it will be reset when we exit the zenmode) but the things that are changed do save their state permanent by saving it into the storage. That's why the new window thinks centered layout should still be active...

We could fix that (and by fix I mean “make it seem to work”) by changing the storage scope to workspace but as @isidorn pointed out that isn't a good solution because the centered layout should be aligned to the vertical/horizontal setting

I don't see how we can fix this without somehow rewriting the code for the save procedure of all the things zenmode changes... because they should not save their state if they are only temporarily changed because of the zenmode

If we want to address this (and I think we should) it could also be the solution for what I mentioned in #45773... luckily I haven't started to implement anything :smile:

**Comment by isidorn** [comment]
Nothing for this milestone since we are in endgame week, thus I am assigning it to the april milestone so we can discuss this in greater detail when I am back from vacation in about 2 weeks.

**Comment by isidorn** [comment]
I was on vacation for the previous milestone thus pushing this to May.
I will investigate how to best fix this next week. Suggestions are welcome

**Comment by SrTobi** [comment]
This is a very nasty problem... I think a  solution would be to not save the state of the centered layout if it was activated because of transitioning to zen mode.

**Comment by uloco** [comment]
I really like the centered layout feature. Unfortunately the centered view gets lost on diff views, even if inline diff is activated. Having a centered layout here also would make me really happy. :)))

**Comment by SrTobi** [comment]
Yeah, we did that intentionally because we thought that the user would want to use the full editor space when watching two files side by side.

Could you maybe elaborate why/how you are using the centered layout? Is your screen just that wide :smiley:  or is there another usecase here that needs the diff view centered?

**Comment by uloco** [comment]
Using inline diff, you do not have two files side by side as you can see here:

![jun-15-2018 14-44-30](https://user-images.githubusercontent.com/8818340/41468532-9fda6b06-70aa-11e8-98f8-6397ee787605.gif)


So I want to stay in centered mode for ever :)

**Comment by SrTobi** [comment]
Ah ok, makes sense :smile: As I see it, the layout of vscode is currently being reworked and so is the centered mode. It isn't even working on `master` right now. We'll have to wait and see.
